src/channel.rs

```diff
@@ -14,9 +14,11 @@ use crate::reply::ReplySender;
 
 /// Helper function to provide options as a fuse_args struct
 /// (which contains an argc count and an argv pointer)
-fn with_fuse_args<T, F: FnOnce(&fuse_args) -> T>(options: &[&OsStr], f: F) -> T {
-    let mut args = vec![CString::new("rust-fuse").unwrap()];
-    args.extend(options.iter().map(|s| CString::new(s.as_bytes()).unwrap()));
+fn with_fuse_args<T, F: FnOnce(&fuse_args) -> T>(options: &OsStr, f: F) -> T {
+    let args = vec![
+        CString::new("rust-fuse").unwrap(),
+        CString::new(options.to_str().unwrap()).unwrap()
+    ];
```

Comment:
This is a change in behavior: the current code passes FUSE a list of N arguments; this code always gives it only 2 arguments, where everything after the first is combined into one string. This doesn't actually work correctly; if you play with the example a bit, you'll notice that everything past the first space character gets ignored by FUSE.

Reply:
Thanks for the feedback. Ill mark the PR as WIP while I take a look at this.

Reply:
👍 a good way to test this is by passing `--help` as one of the arguments. libfuse will print its usage text if it sees that.